### DISCUSSION TITLE
ts(2037) Intellisense error "Cannot find module or its corresponding type declaration"

### DISCUSSION THREAD

**Discussion by dsriseah** [dsriseah/ursys#33] [00 General]
The issue: Your code compiles, but Intellisense complains about the module not being found.

![image](https://github.com/user-attachments/assets/09eb545f-f8bc-4d1d-a7e8-f190f5819993)

The resolution:

- Did you run the type file builder for ur's `_dist`? Use `ur build` or the `@build-ursys.sh` script
- Does `tsconfig.json` include your source directory? This was the reason for me. For example, if you create a new top-level special folder like `app-*` or `sna-*`, you need to make sure that this is in the `include` paths:

```tsconfig
  "include": [
    "_ur/**/*.ts",
    "_ur/**/*.mts",
    "_ur_addons/**/*.ts",
    "_ur_addons/**/*.mts",
    "sna-*/**.ts",
    "sna-*/**.mts",
    "app-*/**.ts",
    "app-*/**.mts"
  ],
```

**Comment by tanyatiwari995** [comment]
Check if the Module is Installed
Run the following command to verify if the module is installed:
npm list @ursys/core
or
If it's missing, install it using:
npm install @ursys/core


Or if it's a dev dependency:
npm install @ursys/core --save-dev
Build Type Definitions for the Module
The post suggests running the type file builder. If the module provides a build script, run:
ur build
or
./build-ursys.sh
This generates .d.ts type files, which TypeScript needs to recognize the module
 Check tsconfig.json for Include Paths
Make sure your tsconfig.json includes the correct paths. Add this inside tsconfig.json:
"include": [
  "_ur/**/*.ts",
  "ur/**/*.mts",
  "ur_addons/**/*.ts",
  "ur_addons/**/*.mts",
  "sna-*/**/*.ts",
  "sna-*/**/*.mts",
  "app-*/**/*.ts",
  "app-*/**/*.mts"
]
If the @ursys/core module is located inside a custom directory, ensure the path is included.
 Check for Typings in node_modules
Look inside node_modules/@ursys/core/ and check if there is an index.d.ts or types directory.
If it's missing, install the type definitions:


npm install @types/ursys --save-dev
If the package itself is missing .d.ts files, you may need to create a manual type declaration file.

 Restart TypeScript Language Server
After making changes, restart TypeScript:


rm -rf node_modules/.cache
npx tsc --watch
or restart your editor (VS Code).

Check for paths in tsconfig.json
If you're using TypeScript path aliases, verify they are set correctly:

"compilerOptions": {
  "baseUrl": "./",
  "paths": {
    "@ursys/core": ["node_modules/@ursys/core"]
  }
}
Then run:

npx tsc --traceResolution
to check how TypeScript resolves the module.

 Try Importing with Full Path
If all else fails, try using a relative import instead of @ursys/core:
import { UR } from "../../node_modules/@ursys/core";
If this works, then the issue is with TypeScript's module resolution.

Final Step
After making these changes, restart your development server:
npm run dev
